stetho/src/main/java/com/facebook/stetho/inspector/protocol/module/DOM.java

```diff
@@ -196,14 +154,20 @@ public ResolveNodeResponse resolveNode(JsonRpcPeer peer, JSONObject params)
   public void setAttributesAsText(JsonRpcPeer peer, JSONObject params) {
     final SetAttributesAsTextRequest request = mObjectMapper.convertValue(
         params, SetAttributesAsTextRequest.class);
-    final Object object = mObjectIdMapper.getObjectForId(request.nodeId);
 
-    mDOMProvider.postAndWait(new Runnable() {
-      @Override
-      public void run() {
-        mDOMProvider.setAttributesAsText(object, request.text);
-      }
-    });
+    final Object element = mDocument.getElementForNodeId(request.nodeId);
```

Comment:
Why was the implementation of `setAttributesAsText` completely changed? It's not related to what we're doing here. Plus, this new code here in `DOM` is incorrect! All these methods on `mDocument` need to be called on the thread dictated by its `ThreadBound` implementation. We should really put this all back the way it was unless there's a really good reason for all this.